packages/client/hooks/useSpotlightSimulatedDrag.tsx

```diff
@@ -48,6 +50,11 @@ const useSpotlightSimulatedDrag = (
       dropTargetId: null,
       dragId: dragIdRef.current
     })
+    SendClientSegmentEventMutation(atmosphere, 'Closed Spotlight', {
```

Comment:
curious, what can we learn from watching the close events? if we're tracking how long the spotlight is open, should we attach an eventId so we can match the open to the close?

Reply:
Exactly, my thinking was that we could track how long the Spotlight has been open for. I thought that we could identify how long it has been opened for by using the `viewerId` and `reflectionId`, but adding an `eventId` might make life easier. I've pushed an update using the `dragId` for the `eventId`.

Reply:
-1 those event IDs are going to get reset on refresh
either we need a globally unique event id or save when `open spotlight` gets called & then when this gets called you can determine the open duration. i'd probably opt for the latter to make it easy for marketing folks to see the duration

Reply:
> save when open spotlight gets called

Ah did you see the 'Opened Spotlight' event [here](https://github.com/ParabolInc/parabol/blob/d34564218947580b3430066469e822dc710a23af/packages/client/hooks/useAnimatedSpotlightSource.ts#L57)?

My original thinking was the marketing folks could use the `viewerId`, `reflectionId`, and `createdAt` (provided by Segment) fields in the 'Opened Spotlight' event and compare it with the subsequent 'Closed Spotlight' event with the same `viewerId` & `reflectionId`. I'm not sure the `eventId` is required so I've removed that in the latest update.

Also, previously if the user opened the Spotlight and refreshed the page, there would be an 'Opened Spotlight' event but not a 'Closed Spotlight' event. That could muddy the data so I've added a `window.onbeforeunload` function that closes the portal causing the 'Closed Spotlight' event to happen.